tests/testthat/test-concRespCore.R

```diff
@@ -15,3 +15,93 @@ test_that("concRespCore works", {
+  ## (i.e. concentration that is below the fitted EC50 value for a chemical is removed.)
+
+  ## dropping higher concentrations for EPAPLT0617C02
+  htpp_global_input = subset(htpp_global_input, htpp_global_input$chem_id !=  "EPAPLT0617C02" |
+                      htpp_global_input$conc <= 2.98 )
+
+  ## One can use the following code to verify that for curve fitting, EPAPLT0617C02 uses
+  ## 5 unique concentrations, with the highest concentration being 2.98.
+  ## htpp_global_subset[htpp_global_subset$chem_id == "EPAPLT0617C02",c("chem_id", "min_conc", "max_conc","n_conc", "conc")]
+  ## unique(htpp_global_input[htpp_global_input$chem_id == "EPAPLT0617C02","conc"])
+
+
+  my_global <- NULL
+  for (this.chem in unique(htpp_global_subset$chem_id)){
+    ## for each chemical, prepare data for concRespCore
+    temp <- htpp_global_input |> filter(chem_id == this.chem)
+    ## prepare "row" list object per Description
+    row <- list(
+      pg_id = unique(temp$pg_id),
+      stype = unique(temp$stype),
+      chem_id = this.chem,
+      min_conc = min(temp$conc),
+      max_conc = max(temp$conc),
+      n_conc = length(unique(temp$conc)),
+      ctr_mean = CONTROL_GMAH$BMED,
+      ctr_sd = CONTROL_GMAH$CUTOFF,
+      conc = temp$conc,
+      resp = temp$d,
+      bmed = CONTROL_GMAH$BMED,
+      cutoff = CONTROL_GMAH$CUTOFF,
+      onesd = CONTROL_GMAH$ONESD,
+      approach = "global",
+      endpoint = "global"
+    )
+    ## run concRespCore on the ‘row’ list object
+    newLine <- try(concRespCore(row, fitmodels = c("cnst", "hill",  "poly1", "poly2", "pow", "exp2", "exp3","exp4", "exp5"), conthits = T, aicc = F, force.fit = FALSE, bidirectional = FALSE, AUC = FALSE))
+    if(is.null(newLine) | class(newLine)=="try-error"){
+      newLine <- try(concRespCore(row, conthits = F, aicc = FALSE, force.fit = FALSE,  bidirectional = TRUE, fitmodels=c("cnst"), AUC = FALSE))
+    }else if(class(newLine) == "try-error"){
+      newLine <- row
+    }else{
+      rownames(newLine) <- ""
+    }
+    my_global <- rbind(my_global, newLine)
+  }
+
+  ## Compare results
+  ## Compare BMD, top_over_cutoff, and hit-call
+
+  ## vector operation, Order by chemerty id to make sure we are comparing the same
+  ## chemical
+  my_global<- my_global[order(my_global$chem_id),]
+  htpp_global_subset<- htpp_global_subset[order(htpp_global_subset$chem_id),]
+
+  ## Differences in decimals places are normal rounding errors
+  ## check if the differences in the BMD estimates exceed a threshold value
+  ## BMD could be NA, replace NA with -1 so it wouldn't cause trouble with all()
+  my_global$bmd[is.na(my_global$bmd)] <- -1
+  htpp_global_subset$bmd[is.na(htpp_global_subset$bmd)] <- -1
+  bmd_check <- all(abs(my_global$bmd - htpp_global_subset$bmd) < 1e-2)
+  expect_true(bmd_check)
+
+  # check if the differences in hit-calls exceed a threshold value
+  hitcall_check <- all(abs(my_global$hitcall - htpp_global_subset$hitcall) < 1e-3)
```

Comment:
Similar question about the range of the order of magnitude difference between the continuous hitcall values as before with the BMD estimates -- that is where did the threshold of 1e-3 come from and is it reasonable.